docs/manual/developer/03_creating_content.md

```diff
@@ -355,3 +355,372 @@ A profile should define these attributes:
+`yes`, `no`, `partially`. The `automated` key is just for informational purposes
+and does not have any impact on the processing.
+
+When XCCDF rules exist, we will assign them to the controls. We will distinguish
+between XCCDF rules which directly implement the given controls (represented by
+`rules` YAML key) and rules that are only related or relevant to the control
+(represented by `related_rules` YAML key).
+
+The `rules` and `related_rules` keys consist of a list of rule IDs. If a rule
+needs to be chosen only in some of products despite its `prodtype` we can use
+Jinja macros inside the controls file to choose products.
+
+After we finish our analysys, we will insert our findings to the controls file,
+the file will look like this:
+
+```
+$ cat controls/abcd.yml
+ 
+id: abcd
+title: ABCD Benchmark for securing Linux systems
+version: 1.2.3
+source: https://www.abcd.com/linux.pdf
+controls:
+  - id: R1
+    title: User session timeout
+    description: |-
+      Remote user sessions must be closed after a certain
+      period of inactivity.
+    automated: yes
+    rules:
+    - sshd_set_idle_timeout
+    - accounts_tmout
+    - var_accounts_tmout=10_min
+{{% if product == “rhel9” %}}
+    - cockpit_session_timeout
+{{% endif %}}
+  - id: R2
+    title: Minimization of configuration
+    description: |-
+      The features configured at the level of launched services
+      should be limited to the strict minimum.
+    automated: no
+    note: |-
+      This is individual depending on the system workload
+      therefore needs to be audited manually.
+    related_rules:
+       - systemd_target_multi_user
+  - id: R3
+    title: Enabling SELinux targeted Policy
+    description: |-
+      It is recommended to enable SELinux in enforcing mode
+      and to use the targeted policy.
+    automated: yes
+```
+
+Notice that the `rules` key in control R1 references the rules. Therefore we can
+use the controls YAML file to assign references instead of adding references to
+each `rule.yml` file. This isn't implemented at the moment.
+
+
+### Defining levels
+
+Some real world policies, eg. ANSSI, have a concept of levels. This means that
+for some use cases a certain set of requirements is required and for other use
+cases a superset of the previous set is required.
+
+Control files can work with the levels concept.
+
+For example, let's say that ABCD benchmark would define 2 levels: low and high.
+The low level would contain R1 and R2. The high level would contain everything
+from the low level and R3, ie. the high level would contain R1, R2 and R3.
+
+First, add the `levels` key to the YAML file and list the level names from
+lowest to highest. The tools working with the controls file assume that every
+control included in a level are also included in all subseqent levels listed in
+the `levels` list.
+
+Second, add `level` key to every control ID to specify level the control belongs to.
```

Comment:
Completely out-of-scope to this PR but CIS is organized as follows:

- Level 1 Server
    - Level 2 Server inheriting all of Level 1 Server's rules
- Level 1 Workstation
    - Level 2 Workstation inheriting all of Level 1 Workstation's rules

Notably, there are two separate hierarchy of levels, or rather, that there's two separate "tags" (workstation, server) that are completely independent.

How would this be represented in a control file? My understanding is we'd need two separate control files (one for Workstation one for Server), which limits the amount of reuse between them.

Could levels be converted to an arbitrary hierarchy? Something like:

```yaml
levels:
 - low
 - medium
 - high
```
for a simple hierarchy and:

```yaml
levels:
 - id: l1-server
 - id: l1-workstation
 - id: l2-server
   inherit:
     - l1-server
 - id: l2-workstation
   inherit:
     - l1-workstation
```

For a custom-defined hierarchy with arbitrary nesting?

Then `level` in the section identifier itself would also have to take a list.


Just a thought.

Reply:
Yes, it would make sense. It's a great idea.